File: php/src/SimpleNumber.php
<?php
declare(strict_types=1);

final class SimpleNumber {
    private $x;

    public function __construct($num) {
        if (!(is_float($num) || is_int($num))) {
            throw new InvalidArgumentException(
                sprintf(
                    '"%s" is not a valid type, must be either a float or int',
                    $num
                )
            );
        }

        $this->x = $num;
    }

    public function add($y) {
        return $this->x + $y;
    }

    public function multiply($y) {
        return $this->x * $y;
    }
}

File: go/simple_number.go
package main

type SimpleInt struct {
	x int
}

func (sn SimpleInt) add(num int) int {
	return sn.x + num
}

func (sn SimpleInt) multiply(num int) int {
	return sn.x * num
}

type Numeric interface {
	add(num int) int
	multiply(num int) int
}

func main() {

}
